pac4j-jwt/src/main/java/org/pac4j/jwt/profile/JwtGenerator.java

```diff
@@ -67,22 +67,34 @@ public JwtGenerator(final String signingSecret, final String encryptionSecret) {
         this.signingSecret = signingSecret;
         this.encryptionSecret = encryptionSecret;
     }
-
+    
     /**
      * Generates a JWT from a user profile.
      *
      * @param profile the given user profile
      * @return the created JWT
      */
     public String generate(final U profile) {
+    	return generate(profile, null, this.jwsAlgorithm);
```

Comment:
Indeed, in the `signJwt` method (line 133), the test is correct. Though, the problem is that this method will never be called because of the test line 94: you should not test if the `signer` is `null`, but if the `signingSecret` is not blank.